src/pages/blog/http-and-lambda.mdx

```diff
@@ -0,0 +1,338 @@
+---
+author: 'osm'
+layout: '../../layouts/BlogPost.astro'
+title: 'Calling AWS Lambda over HTTP'
+description: 'Everything You Need to Know About Calling AWS Lambda over HTTP: Setup, Formats, and Error Handling Explained'
+category: 'architecture'
+publishedDate: '2024-06-11'
+tags:
+  - http
+  - aws
+  - lambda
+  - 'api gateway'
+  - cloudfront
+heroImage: 'xt24-stage.jpg'
+---
+
+{/*
+
+The requirement that this blog post fulfils is:
+As a developer, I want to know:
+- What different ways are there I can call a lambda over HTTP
+- For each of those:
+  - What is the formats of requests and responses?
+  - What happens if the request/response is in the wrong format?
+  - What happens if exceptions are thrown?
+  - What other quirks are there?
+And I want to have it all in one place.
+
+*/}
+
+There are three different methods to call an AWS Lambda over HTTP externally:
+[Function URLs](https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html),
+[Application Load Balancers](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html)
+(ALBs)
+and [API Gateway](https://aws.amazon.com/api-gateway/)
+
+As I investigated them I found that each had slightly different behaviours,
+quirks and setup instructions. I ended up with 20+ tabs open and writing a
+bunch of test cases, so I thought I would share what I had learned all in one
+place.
+
+
+
+## Methodology
+
+While researching, the questions tried to answer were:
+1. How do I set this up?
+2. What are the input and output formats?
+3. What happens if I don't use these formats?
+4. What happens if an exception is thrown?
+5. What other quirks are there?
+
+For the most part I can link you to the relevant part of the docs, but for
+some I needed to run some tests. The tests I ran were:
```

Comment:
'run ~~some tests~~ a few experiments'? again to avoid repetition of 'some', but also 'test' has a more specific meaning that you probably don't mean to imply here